lib/connectors_shared/constants.rb

```diff
@@ -18,7 +20,7 @@ class Constants
     # full, it will sleep for maximum MAX_IDDLE_ATTEMPTS times, and if the queue is still
     # full, then job will be terminated.
     JOB_QUEUE_SIZE_IDLE_THRESHOLD = 500 # How many documents the job queue stores until it sleeps
-    IDLE_SLEEP_TIME = 10 # For how long job queue will sleep before checking the queue size again
-    MAX_IDLE_ATTEMPTS = 30 # How many consecutive times job will try to sleep until it's destroyed
+    IDLE_SLEEP_TIME = 10.seconds # For how long job queue will sleep before checking the queue size again
+    MAX_IDLE_TIME = 300.seconds # How long a job will try to sleep until it's destroyed
```

Comment:
Make the limit higher:
```suggestion
    MAX_IDLE_TIME = 600.seconds # How long a job will try to sleep until it's destroyed
```

Or:
```suggestion
    MAX_IDLE_TIME = 10.minutes # How long a job will try to sleep until it's destroyed
```